app/controlplane/internal/biz/referrer.go

```diff
@@ -37,27 +39,21 @@ type Referrer struct {
 	Kind   string
 	// Wether the item is downloadable from CAS or not
 	Downloadable bool
-	// points to other digests
-	References []string
+	References   []*Referrer
 }
 
 // Actual referrer stored in the DB which includes a nested list of storedReferences
 type StoredReferrer struct {
-	ID     uuid.UUID
-	Digest string
-	Kind   string
-	// Wether the item is downloadable from CAS or not
-	Downloadable bool
-	CreatedAt    *time.Time
+	*Referrer
+	ID        uuid.UUID
```

Comment:
Why do we need uuid?

Reply:
In favor of another format? or having a primary ID at all?